web/src/main/java/se/inera/intyg/webcert/web/service/facade/util/CertificateRecipientConverterImpl.java

```diff
@@ -0,0 +1,32 @@
+package se.inera.intyg.webcert.web.service.facade.util;
+
+import java.time.LocalDateTime;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.common.support.facade.model.metadata.CertificateRecipient;
+import se.inera.intyg.webcert.web.service.receiver.CertificateReceiverService;
+
+@Service
+public class CertificateRecipientConverterImpl implements CertificateRecipientConverter {
+
+  private final CertificateReceiverService certificateReceiverService;
+
+  public CertificateRecipientConverterImpl(CertificateReceiverService certificateReceiverService) {
+    this.certificateReceiverService = certificateReceiverService;
+  }
+
+  @Override
+  public CertificateRecipient get(String type, String certificateId, LocalDateTime sent) {
+    final var recipients = certificateReceiverService.listPossibleReceiversWithApprovedInfo(type, certificateId);
+
+    return recipients
+        .stream()
+        .findFirst()
```

Comment:
listPossibleReceiversWithApprovedInfo can (in test environments) return multiple recipients. Even if the "huvudmottagare" probably/might be the first element, I would suggest to filter out any receiver that isn't "isLocked" ("huvudmottagare") before findFirst().